src/ballet/sha256/fd_sha256.c

```diff
@@ -200,7 +200,19 @@ fd_sha256_core_ref( uint *        state,
 
 }
 
-#define fd_sha256_core fd_sha256_core_ref
+/* Name of internal definition of block function.
+   See fd_sha256_core_ref (reference implementation) for the method signature.
+   block_cnt must be greater than zero. */
+#define fd_sha256_core_ fd_sha256_core_ref
+
+/* Export block function */
+void
+fd_sha256_core( uchar       * FD_RESTRICT state,
```

Comment:
What do you think about renaming this to avoid the name collision that you resolved with the extra `_`? Some ideas: `fd_sha256_chunk`, `fd_sha256_block`, `fd_sha256_hash_blocks`